File: Algorithms/src/744. Find Smallest Letter Greater Than Target/Code/solution.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int left = 0;
        int right = letters.length - 1;
        char answer = letters[0]; // wrap-around default

        while (left <= right) {
            int mid = left + (right - left) / 2;

            if (letters[mid] > target) {
                answer = letters[mid];
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return answer;
    }
}

File: Algorithms/src/744. Find Smallest Letter Greater Than Target/Code/solution.go
func nextGreatestLetter(letters []byte, target byte) byte {
    left := 0
    right := len(letters) - 1
    answer := letters[0] // wrap-around default

    for left <= right {
        mid := left + (right-left)/2

        if letters[mid] > target {
            answer = letters[mid]
            right = mid - 1
        } else {
            left = mid + 1
        }
    }

    return answer
}
